.buildkite/scripts/dra-workflow.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+set -euo pipefail
+
+WORKFLOW="${DRA_WORKFLOW:-snapshot}"
+BRANCH="${BUILDKITE_BRANCH:-}"
+
+# Don't publish main branch to staging
+if [[ "$BRANCH" == "main" && "$WORKFLOW" == "staging" ]]; then
+  exit 0
+fi
+
+echo --- Preparing
+
+# TODO move this to image
+sudo apt-get update -y
+sudo apt-get install -y libxml2-utils python3.10-venv
+
+RM_BRANCH="$BRANCH"
+if [[ "$BRANCH" == "main" ]]; then
+  RM_BRANCH=master
+fi
+
+ES_VERSION=$(grep elasticsearch build-tools-internal/version.properties | sed "s/elasticsearch *= *//g")
+
+VERSION_SUFFIX=""
+if [[ "$WORKFLOW" == "snapshot" ]]; then
+  VERSION_SUFFIX="-SNAPSHOT"
+fi
+
+BEATS_BUILD_ID="$(./.ci/scripts/resolve-dra-manifest.sh beats "$RM_BRANCH" "$ES_VERSION" "$WORKFLOW")"
+ML_CPP_BUILD_ID="$(./.ci/scripts/resolve-dra-manifest.sh ml-cpp "$RM_BRANCH" "$ES_VERSION" "$WORKFLOW")"
+
+LICENSE_KEY_ARG=""
+BUILD_SNAPSHOT_ARG=""
+
+if [[ "$WORKFLOW" == "staging" ]]; then
+  LICENSE_KEY=$(mktemp -d)/license.key
+  # Intentionally leaving this secret here instead of a hook
+  # Since it's getting redirected straight to a file, it's probably even safer to just leave it here rather than stick it in an env var for a short period of time
```

Comment:
Also, this isn't really a "secret". It's the public key used for license validation so it's not actually sensitive.

Reply:
Ah, I should have paid closer attention that it was just the public key, thanks. Changed comment about this